### PR TITLE
Import dart:io directly

### PR DESCRIPTION
**Pull request by tvolkert**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+#### 2.4.0
+
+* Import `dart:io` unconditionally. More recent Dart SDK revisions allow
+  `dart:io` to be imported in a browser context, though if methods are actually
+  invoked, they will fail. This matches well with `package:file`, where users
+  can use the `memory` library and get in-memory implementations of the
+  `dart:io` interfaces.
```
Comment (goderbauer) [inline CHANGELOG.md]: nit: maybe mention in a extra bullet that the minimum SDK has been bumped to 1.24?

**Comment by keertip** [comment]
Thanks!

File: pubspec.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: file
-version: 2.3.7
+version: 2.4.0
```
Comment (yjbanov) [inline pubspec.yaml]: Seems like a pretty major breaking change. Maybe start `3.0.0`?

Comment (tvolkert) [reply pubspec.yaml]: For users past SDK 1.23, it's actually not breaking at all, but I'm happy to bump it to `3.0.0` regardless.  Done!

**Pull request merged**